



### data is read from BTCPrice.csv
### removed all the duplicates - here for duplicates we will just drop one and keep=First 

### and generate BTCPriceWithoutDuplicates.csv
### merge with BTC_Indicators.csv which is generated by running mainCode.py and generate BTCPriceMergedIndicatorsAndWithoutDuplicates

### dropped the first 54 rows so that there are no null values in any feature and 
in below features -- 
Aroon_up
Aroon_down
Aroon_oscillator
%K-FAST Stochastic indicator
ulcer_index
WilliamsR


0 values are replaces with 0.0001 and then save the data to generated BTCPriceCleaned.csv


In [58]:
import numpy as np
import pandas as pd


In [59]:
from IPython.display import display

pd.options.display.max_columns = None

import matplotlib.pyplot as plt



In [60]:
df =pd.read_csv("BTCPrice.csv")
df

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
3023  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3024  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3025  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3026  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3027  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
3023  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
3024  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
3025  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
3026  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
3027  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
3023  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
3024   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
3025   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
3026  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
3027  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
3023  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
3024  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
3025  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
3026  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
3027  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
3023  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
3024  0.041888  0.045579  0.007892  0

df =pd.read_csv("BTCPrice.csv")
df

In [61]:
import yfinance as yf
data = yf.download(tickers='BTC-USD',start =  '2014-09-17',end='2022-11-23',  interval = '1d')
data

[*********************100%***********************]  1 of 1 completed


Open          High           Low  \
Date                                                                  
2014-09-17 00:00:00+00:00    465.864014    468.174011    452.421997   
2014-09-18 00:00:00+00:00    456.859985    456.859985    413.104004   
2014-09-19 00:00:00+00:00    424.102997    427.834991    384.532013   
2014-09-20 00:00:00+00:00    394.673004    423.295990    389.882996   
2014-09-21 00:00:00+00:00    408.084991    412.425995    393.181000   
...                                 ...           ...           ...   
2022-11-18 00:00:00+00:00  16687.912109  16947.058594  16564.611328   
2022-11-19 00:00:00+00:00  16696.220703  16797.876953  16570.410156   
2022-11-20 00:00:00+00:00  16712.919922  16746.779297  16248.692383   
2022-11-21 00:00:00+00:00  16291.223633  16291.223633  15599.046875   
2022-11-22 00:00:00+00:00  15782.300781  16253.047852  15656.606445   

                                  Close     Adj Close       Volume  
Date                                                                
2014-09-17 00:00:00+00:00    457.334015    457.334015     21056800  
2014-09-18 00:00:00+00:00    424.440002    424.440002     34483200  
2014-09-19 00:00:00+00:00    394.795990    394.795990     37919700  
2014-09-20 00:00:00+00:00    408.903992    408.903992     36863600  
2014-09-21 00:00:00+00:00    398.821014    398.821014     26580100  
...                                 ...           ...          ...  
2022-11-18 00:00:00+00:00  16697.777344  16697.777344  26862218609  
2022-11-19 00:00:00+00:00  16711.546875  16711.546875  16106223492  
2022-11-20 00:00:00+00:00  16291.832031  16291.832031  21313378652  
2022-11-21 00:00:00+00:00  15787.284180  15787.284180  37429485518  
2022-11-22 00:00:00+00:00  16189.769531  16189.769531  30726828760  

[2989 rows x 6 columns]

In [62]:
data.reset_index(inplace=True)
data.head()

Date        Open        High         Low       Close  \
0 2014-09-17 00:00:00+00:00  465.864014  468.174011  452.421997  457.334015   
1 2014-09-18 00:00:00+00:00  456.859985  456.859985  413.104004  424.440002   
2 2014-09-19 00:00:00+00:00  424.102997  427.834991  384.532013  394.795990   
3 2014-09-20 00:00:00+00:00  394.673004  423.295990  389.882996  408.903992   
4 2014-09-21 00:00:00+00:00  408.084991  412.425995  393.181000  398.821014   

    Adj Close    Volume  
0  457.334015  21056800  
1  424.440002  34483200  
2  394.795990  37919700  
3  408.903992  36863600  
4  398.821014  26580100

In [63]:
def retDate(x):
    y=x.strftime("%d%b%Y").lower()
    return y


data['dateFormatted'] = data['Date'].apply(retDate)

In [64]:
data.head()

Date        Open        High         Low       Close  \
0 2014-09-17 00:00:00+00:00  465.864014  468.174011  452.421997  457.334015   
1 2014-09-18 00:00:00+00:00  456.859985  456.859985  413.104004  424.440002   
2 2014-09-19 00:00:00+00:00  424.102997  427.834991  384.532013  394.795990   
3 2014-09-20 00:00:00+00:00  394.673004  423.295990  389.882996  408.903992   
4 2014-09-21 00:00:00+00:00  408.084991  412.425995  393.181000  398.821014   

    Adj Close    Volume dateFormatted  
0  457.334015  21056800     17sep2014  
1  424.440002  34483200     18sep2014  
2  394.795990  37919700     19sep2014  
3  408.903992  36863600     20sep2014  
4  398.821014  26580100     21sep2014

In [65]:
# checking if there are duplicate records for same date

In [66]:
pd.options.display.max_rows = None

print(df['date'].value_counts() >1)
pd.options.display.max_rows = 10

12-Oct-22     True
8-Oct-22      True
21-Oct-22     True
20-Oct-22     True
19-Oct-22     True
18-Oct-22     True
17-Oct-22     True
16-Oct-22     True
15-Oct-22     True
14-Oct-22     True
13-Oct-22     True
11-Oct-22     True
10-Oct-22     True
9-Oct-22      True
7-Oct-22      True
23-Oct-22     True
6-Oct-22      True
5-Oct-22      True
4-Oct-22      True
3-Oct-22      True
2-Oct-22      True
1-Oct-22      True
30-Sep-22     True
29-Sep-22     True
28-Sep-22     True
27-Sep-22     True
26-Sep-22     True
25-Sep-22     True
22-Oct-22     True
24-Oct-22     True
24-Sep-22     True
5-Nov-22      True
26-Oct-22     True
27-Oct-22     True
28-Oct-22     True
29-Oct-22     True
30-Oct-22     True
31-Oct-22     True
25-Oct-22     True
24-Feb-20    False
25-Feb-20    False
26-Feb-20    False
1-Mar-20     False
27-Feb-20    False
28-Feb-20    False
29-Feb-20    False
22-Feb-20    False
2-Mar-20     False
23-Feb-20    False
10-Feb-20    False
21-Feb-20    False
20-Feb-20    False
19-Feb-20   

In [67]:
# there are duplicates for these records

pd.options.display.max_rows = None
duplicate = df[df.duplicated()]
duplicate


id assetcode  \
2944  mp:2022-10-01_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2946  mp:2022-10-02_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2948  mp:2022-10-03_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2950  mp:2022-10-04_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2952  mp:2022-10-05_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2954  mp:2022-10-06_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2956  mp:2022-10-07_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2958  mp:2022-10-08_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2960  mp:2022-10-09_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2962  mp:2022-10-10_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2964  mp:2022-10-11_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2966  mp:2022-10-12_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2968  mp:2022-10-13_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2970  mp:2022-10-14_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2972  mp:2022-10-15_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2974  mp:2022-10-16_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2976  mp:2022-10-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2978  mp:2022-10-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2980  mp:2022-10-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2982  mp:2022-10-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2984  mp:2022-10-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-10-22_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-10-23_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2990  mp:2022-10-24_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2992  mp:2022-10-25_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2994  mp:2022-10-26_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2996  mp:2022-10-27_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2998  mp:2022-10-28_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3000  mp:2022-10-29_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3002  mp:2022-10-30_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3004  mp:2022-10-31_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
2944  2022-10-01T19:30:00.000Z  News_Social     MP:4.0.20     24923   
2946  2022-10-02T19:30:00.000Z  News_Social     MP:4.0.20     21817   
2948  2022-10-03T19:30:00.000Z  News_Social     MP:4.0.20     30805   
2950  2022-10-04T19:30:00.000Z  News_Social     MP:4.0.20     32567   
2952  2022-10-05T19:30:00.000Z  News_Social     MP:4.0.20     29994   
2954  2022-10-06T19:30:00.000Z  News_Social     MP:4.0.20     31846   
2956  2022-10-07T19:30:00.000Z  News_Social     MP:4.0.20     29786   
2958  2022-10-08T19:30:00.000Z  News_Social     MP:4.0.20     22772   
2960  2022-10-09T19:30:00.000Z  News_Social     MP:4.0.20     21958   
2962  2022-10-10T19:30:00.000Z  News_Social     MP:4.0.20     28757   
2964  2022-10-11T19:30:00.000Z  News_Social     MP:4.0.20     32012   
2966  2022-10-12T19:30:00.000Z  News_Social     MP:4.0.20     30560   
2968  2022-10-13T19:30:00.000Z  News_Social     MP:4.0.20     34377   
2970  2022-10-14T19:30:00.000Z  News_Social     MP:4.0.20     31740   
2972  2022-10-15T19:30:00.000Z  News_Social     MP:4.0.20     24879   
2974  2022-10-16T19:30:00.000Z  News_Social     MP:4.0.21     22018   
2976  2022-10-17T19:30:00.000Z  News_Social     MP:4.0.21     30173   
2978  2022-10-18T19:30:00.000Z  News_Social     MP:4.0.21     29181   
2980  2022-10-19T19:30:00.000Z  News_Social     MP:4.0.21     30023   
2982  2022-10-20T19:30:00.000Z  News_Social     MP:4.0.21     31068   
2984  2022-10-21T19:30:00.000Z  News_Social     MP:4.0.21     29605   
2986  2022-10-22T19:30:00.000Z  News_Social     MP:4.0.21     20936   
2988  2022-10-23T19:30:00.000Z  News_Social     MP:4.0.21     21872   
2990  2022-10-24T19:30:00.000Z  News_Social     MP:4.0.21     29986   
2992  2022-10-25T19:30:00.000Z  News_Social     MP:4.0.21     30864  

In [68]:
pd.options.display.max_rows = 10


In [69]:
df.drop_duplicates(inplace=True)

In [70]:
df.reset_index(inplace=True)
df.drop('index',axis=1,inplace=True)
df

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2992  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2993  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2994  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2995  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2996  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2992  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2993  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2994  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2995  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2996  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2992  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2993   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2994   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2995  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2996  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2992  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2993  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2994  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2995  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2996  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2992  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2993  0.041888  0.045579  0.007892  0

# still there are multiple records for same dates.


In [71]:
pd.options.display.max_rows = None
print(print(df['date'].value_counts() >1))
pd.options.display.max_rows = 10

30-Sep-22     True
29-Sep-22     True
28-Sep-22     True
27-Sep-22     True
26-Sep-22     True
25-Sep-22     True
24-Sep-22     True
5-Nov-22      True
17-Sep-14    False
2-Mar-20     False
25-Feb-20    False
26-Feb-20    False
27-Feb-20    False
28-Feb-20    False
29-Feb-20    False
1-Mar-20     False
6-Mar-20     False
3-Mar-20     False
4-Mar-20     False
5-Mar-20     False
23-Feb-20    False
7-Mar-20     False
8-Mar-20     False
9-Mar-20     False
24-Feb-20    False
22-Feb-20    False
11-Mar-20    False
21-Feb-20    False
31-Jan-20    False
1-Feb-20     False
2-Feb-20     False
3-Feb-20     False
4-Feb-20     False
5-Feb-20     False
6-Feb-20     False
7-Feb-20     False
8-Feb-20     False
9-Feb-20     False
10-Feb-20    False
11-Feb-20    False
12-Feb-20    False
13-Feb-20    False
14-Feb-20    False
15-Feb-20    False
16-Feb-20    False
17-Feb-20    False
18-Feb-20    False
19-Feb-20    False
20-Feb-20    False
10-Mar-20    False
12-Mar-20    False
29-Jan-20    False
6-Apr-20    

In [72]:
df_copy=df.copy()

# lets do df.groupby('id').mean().reset_index()


In [73]:
df.drop_duplicates(subset='id', keep='first',inplace=True)


In [74]:
df

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2992  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2993  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2994  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2995  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2996  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2992  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2993  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2994  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2995  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2996  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2992  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2993   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2994   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2995  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2996  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2992  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2993  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2994  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2995  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2996  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2992  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2993  0.041888  0.045579  0.007892  0

there are 2989 rows in the data downloaded from yahoofinance but there is 2990 here that means still there is one extra

In [75]:
def partOfDate(x):
    return x[:13]
df['partOfDate']=df['id'].apply(partOfDate)

In [76]:

print(df['partOfDate'].value_counts() >1)


mp:2022-11-05     True
mp:2014-09-17    False
mp:2020-02-22    False
mp:2020-02-24    False
mp:2020-02-25    False
                 ...  
mp:2017-06-12    False
mp:2017-06-13    False
mp:2017-06-14    False
mp:2017-06-15    False
mp:2022-11-22    False
Name: partOfDate, Length: 2989, dtype: bool


In [77]:
df[df.duplicated('partOfDate')]['partOfDate']

2979    mp:2022-11-05
Name: partOfDate, dtype: object

In [78]:
duplicateDates = df[df.duplicated('partOfDate')]['partOfDate'].array
def retPresence(x):
    return x in duplicateDates

df[df['partOfDate'].apply(retPresence)]
#in ['17sep2014']

id assetcode  \
2978  mp:2022-11-05_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2979  mp:2022-11-05_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
2978  2022-11-05T20:30:00.000Z  News_Social     MP:4.0.21      1136   
2979  2022-11-05T19:30:00.000Z  News_Social     MP:4.0.21     28879   

           buzz  sentiment  negative  positive  optimism  pessimism       joy  \
2978   4070.750   0.035497  0.205736  0.241233  0.054535   0.061659  0.007984   
2979  97707.852   0.013208  0.207721  0.220929  0.058736   0.051751  0.008229   

      lovehate     trust     anger  disagreement      fear     gloom  \
2978  0.006141  0.003685  0.016459      0.034515  0.003808  0.027022   
2979  0.002994  0.005757  0.012322      0.031446  0.004565  0.029501   

        stress  violence  surprise  timeurgency  uncertainty  emotionvsfact  \
2978  0.029847  0.005282  0.003562    -0.001965     0.026089       0.395234   
2979  0.037034  0.008229  0.004473     0.001878     0.023093       0.352634   

      shortvslongterm  marketrisk  longshort  longshortforecast  \
2978         0.006633    0.003316  -0.001720           0.002457   
2979         0.013868    0.027367   0.001735           0.000757   

      pricedirection  pricedown  priceforecast   priceup  pricemomentum  \
2978        0.013265   0.022846      -0.001842  0.036111       0.002702   
2979        0.021554   0.021933       0.001591  0.043487       0.001412   

      topvsbottom  overvaluedvsundervalued  volatility  adoption  \
2978    -0.005896                -0.000123    0.014555  0.004913   
2979     0.000578                -0.000307    0.017023  0.006069   

      adoptionforecast  anonymity    attack  codesentiment  codeupgrade  \
2978          0.000737   0.001474  0.000737       0.000614     0.002211   
2979          0.001034   0.000983  0.000527       0.000353     0.001755   

      criminalactivity  developersentiment      fomo      fork  forkforecast  \
2978          0.000491            0.000737  0.001965  0.000491      0.000454   
2979          0.001244            0.000015  0.000527  0.000399      0.000113   

          hodl  innovation  litigation  majorityattack     noobs  \
2978  0.001965    0.001965    0.000737        0.000210  0.000737   
2979  0.001545    0.000952    0.000353        0.000046  0.000578   

      regulatorycrackdown  regulatoryissues      scam  transactionspeed  \
2978             0.001228          0.004422  0.005650          0.000246   
2979             0.000629          0.003864  0.002861          0.000440   

      vulnerability  partnership  futurevspast windowtimestamp1  \
2978       0.000123     0.000491     -0.030154        11/5/2022   
2979       0.000425     0.000235     -0.066399        11/5/2022   

     windowtimestamp2      date        close  r_BTC_USD  R_BTC_USD  \
2978    20:30:00.000Z  5-Nov-22  21282.69141   0.006385   0.006406   
2979    19:30:00.000Z  5-Nov-22  21282.69141   0.006385   0.006406   

      volume_BTC_USD     partOfDate  
2978    3.784605e+10  mp:2022-11-05  
2979    3.784605e+10  mp:2022-11-05

there are 2 records for 2022-11-05 one for 19.30 time and other for 20.30 time
dropping the second record

In [79]:
df = df.drop_duplicates(subset='partOfDate', keep='last')
df.reset_index(inplace=True)
df.drop('index',axis=1,inplace=True)
df

C:\Users\Student\AppData\Local\Temp\ipykernel_7772\4000680985.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.drop('index',axis=1,inplace=True)


id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2984  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0

In [80]:
df.drop(['partOfDate'],axis=1,inplace=True)

C:\Users\Student\AppData\Local\Temp\ipykernel_7772\3079855058.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df.drop(['partOfDate'],axis=1,inplace=True)


In [81]:
df.to_csv("BTCPriceWithoutDuplicates.csv")

### merge BTC_Indicators.csv and BTCPriceWithoutDuplicates.csv

In [82]:
df_Indicators = pd.read_csv("BTC_Indicators.csv",index_col = 'Unnamed: 0')
df_Indicators

Date          Open          High           Low         Close  \
0     2014-09-17    465.864014    468.174011    452.421997    457.334015   
1     2014-09-18    456.859985    456.859985    413.104004    424.440002   
2     2014-09-19    424.102997    427.834991    384.532013    394.795990   
3     2014-09-20    394.673004    423.295990    389.882996    408.903992   
4     2014-09-21    408.084991    412.425995    393.181000    398.821014   
...          ...           ...           ...           ...           ...   
2984  2022-11-18  16687.912109  16947.058594  16564.611328  16697.777344   
2985  2022-11-19  16696.220703  16797.876953  16570.410156  16711.546875   
2986  2022-11-20  16712.919922  16746.779297  16248.692383  16291.832031   
2987  2022-11-21  16291.223633  16291.223633  15599.046875  15787.284180   
2988  2022-11-22  15782.300781  16253.047852  15656.606445  16189.769531   

         Adj Close       Volume dateFormatted  MoneyFlowVolume  \
0       457.334015     21056800     17sep2014    -7.924336e+06   
1       424.440002     34483200     18sep2014    -1.661586e+07   
2       394.795990     37919700     19sep2014    -1.994371e+07   
3       408.903992     36863600     20sep2014     5.107041e+06   
4       398.821014     26580100     21sep2014    -1.100076e+07   
...            ...          ...           ...              ...   
2984  16697.777344  26862218609     18nov2022    -8.155668e+09   
2985  16711.546875  16106223492     19nov2022     3.880689e+09   
2986  16291.832031  21313378652     20nov2022    -1.762145e+10   
2987  15787.284180  37429485518     21nov2022    -1.707161e+10   
2988  16189.769531  30726828760     22nov2022     2.420702e+10   

      accumulation_distribution_indicator  Aroon_up  Aroon_down  \
0                           -7.924336e+06       NaN         NaN   
1                           -2.454020e+07       NaN         NaN   
2                           -4.448391e+07       NaN         NaN   
3                           -3.937687e+07       NaN         NaN   
4                           -5.037763e+07       NaN         NaN   
...                                   ...       ...         ...   
2984                         4.006859e+12      48.0        64.0   
2985                         4.010740e+12      44.0        60.0   
2986                         3.993118e+12      40.0        56.0   
2987                         3.976047e+12      36.0       100.0   
2988                         4.000254e+12      32.0        96.0   

      Aroon_oscillator      Adj Open      Adj High       Adj Low  \
0                  NaN    465.864014    468.174011    452.421997   
1                  NaN    456.859985    456.859985    413.104004   
2                  NaN    424.102997    427.834991    384.532013   
3                  NaN    394.673004    423.295990    389.882996   
4                  NaN    408.084991    412.425995    393.181000   
...                ...           ...           ...           ...   
2984             -16.0  16687.912109  16947.058594  16564.611328   
2985             -16.0  16696.220703  16797.876953  16570.410156   
2986             -16.0  16712.919922  16746.779297  16248.692383   
2987             -64.0  16291.223633  16291.223633  15599.046875   
2988             -64.0  15782.300781  16253.047852  15656.606445   

      adx_pos_directional_indicator  adx_neg_directional_indicator  \
0                          0.000000                       0.000000   
1                          0.000000                       0.000000   
2                          0.000000                       0.000000   
3                          0.000000                       0.000000   
4                          0.000000                       0.000000   
...                             ...                            ...   
2984                      10.126249                      33.000040   
2985                       9.910023                      32.295389   
2986                       9.434939                   

In [83]:
df_withoutDuplciates = pd.read_csv("BTCPriceWithoutDuplicates.csv",index_col = 'Unnamed: 0')
df_withoutDuplciates

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2984  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0

In [84]:
df_withoutDuplciates['windowtimestamp1']

0        9/17/2014
1        9/18/2014
2        9/19/2014
3        9/20/2014
4        9/21/2014
           ...    
2984    11/18/2022
2985    11/19/2022
2986    11/20/2022
2987    11/21/2022
2988    11/22/2022
Name: windowtimestamp1, Length: 2989, dtype: object

In [85]:
import datetime


In [86]:
def retDate(x):
    y=datetime.datetime.strptime(x, "%m/%d/%Y").date().strftime("%d%b%Y").lower()
    return y


df_withoutDuplciates['dateFormatted'] = df_withoutDuplciates['windowtimestamp1'].apply(retDate)

In [87]:
df_withoutDuplciates['dateFormatted']

0       17sep2014
1       18sep2014
2       19sep2014
3       20sep2014
4       21sep2014
          ...    
2984    18nov2022
2985    19nov2022
2986    20nov2022
2987    21nov2022
2988    22nov2022
Name: dateFormatted, Length: 2989, dtype: object

In [88]:
df_withoutDuplciates

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2984  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0

In [89]:
df_withoutDuplciates['dateFormatted']

0       17sep2014
1       18sep2014
2       19sep2014
3       20sep2014
4       21sep2014
          ...    
2984    18nov2022
2985    19nov2022
2986    20nov2022
2987    21nov2022
2988    22nov2022
Name: dateFormatted, Length: 2989, dtype: object

In [90]:
df_Indicators['dateFormatted']

0       17sep2014
1       18sep2014
2       19sep2014
3       20sep2014
4       21sep2014
          ...    
2984    18nov2022
2985    19nov2022
2986    20nov2022
2987    21nov2022
2988    22nov2022
Name: dateFormatted, Length: 2989, dtype: object

In [91]:
df_merge_Indicators_withoutDuplciates = pd.merge(df_withoutDuplciates, df_Indicators, how ='left', on ='dateFormatted')

In [92]:
df_merge_Indicators_withoutDuplciates

id assetcode  \
0     mp:2014-09-17_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1     mp:2014-09-18_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2     mp:2014-09-19_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3     mp:2014-09-20_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4     mp:2014-09-21_19.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-09-17T19:30:00.000Z  News_Social      MP:4.0.0      1403   
1     2014-09-18T19:30:00.000Z  News_Social      MP:4.0.0      1406   
2     2014-09-19T19:30:00.000Z  News_Social      MP:4.0.0      1704   
3     2014-09-20T19:30:00.000Z  News_Social      MP:4.0.0      1186   
4     2014-09-21T19:30:00.000Z  News_Social      MP:4.0.0       937   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

             buzz  sentiment  negative  positive  optimism  pessimism  \
0       4294.2998   0.117598  0.195142  0.312740  0.065785   0.038307   
1       3564.8501   0.036607  0.217821  0.254429  0.056244   0.056805   
2       5444.1499  -0.073933  0.273229  0.199296  0.052625   0.034349   
3       3603.1499  -0.046210  0.245896  0.199686  0.052038   0.032610   
4       1590.7000   0.028604  0.209970  0.238574  0.063180   0.070409   
...           ...        ...       ...       ...       ...        ...   
2984  116049.5000  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.3520  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.2970  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.4500  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.1600  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.008267  0.013390  0.014671  0.006986      0.026314  0.004075   
1     0.007153  0.004488  0.013745  0.007293      0.013325  0.005190   
2     0.007807  0.003214 -0.013409  0.007623      0.023052  0.004408   
3     0.004996  0.001388 -0.028031  0.010824      0.015819  0.005967   
4     0.005344  0.003143 -0.001886  0.004086      0.022632  0.005972   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.010945  0.034464  0.010712  0.001747     0.020958     0.018303   
1     0.018935  0.035065  0.009117  0.003647     0.020337     0.024657   
2     0.029206  0.039308  0.006245  0.002663    -0.000643     0.025661   
3     0.026505  0.047736  0.004996  0.002498     0.006383     0.026754   
4     0.017602  0.038034  0.005029  0.003458     0.005029     0.038536   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0

In [93]:
df_merge_Indicators_withoutDuplciates.to_csv("BTCPriceMergedIndicatorsAndWithoutDuplicates.csv")

### removing volatility feature and remove the first 54 rows so that there wll be no nulls

In [94]:
df_merge_Indicators_withoutDuplciates_1= df_merge_Indicators_withoutDuplciates.copy()

In [95]:
df_merge_Indicators_withoutDuplciates_1.drop("volatility/sd",axis=1,inplace=True)

In [96]:
pd.options.display.max_rows = None

print(df_merge_Indicators_withoutDuplciates_1.isnull().sum())
pd.options.display.max_rows = 10



id                                            0
assetcode                                     0
windowtimestamp                               0
datatype                                      0
systemversion                                 0
mentions                                      0
buzz                                          0
sentiment                                     0
negative                                      0
positive                                      0
optimism                                      0
pessimism                                     0
joy                                           0
lovehate                                      0
trust                                         0
anger                                         0
disagreement                                  0
fear                                          0
gloom                                         0
stress                                        0
violence                                

In [97]:
pd.options.display.max_rows = None

print(df_merge_Indicators_withoutDuplciates_1.iloc[54:2989].isnull().sum())
pd.options.display.max_rows = 10


id                                           0
assetcode                                    0
windowtimestamp                              0
datatype                                     0
systemversion                                0
mentions                                     0
buzz                                         0
sentiment                                    0
negative                                     0
positive                                     0
optimism                                     0
pessimism                                    0
joy                                          0
lovehate                                     0
trust                                        0
anger                                        0
disagreement                                 0
fear                                         0
gloom                                        0
stress                                       0
violence                                     0
surprise     

In [98]:
df_merge_Indicators_withoutDuplciates_1[52:]

id assetcode  \
52    mp:2014-11-08_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
53    mp:2014-11-09_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
54    mp:2014-11-10_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
55    mp:2014-11-11_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
56    mp:2014-11-12_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
52    2014-11-08T20:30:00.000Z  News_Social      MP:4.0.0       925   
53    2014-11-09T20:30:00.000Z  News_Social      MP:4.0.0       759   
54    2014-11-10T20:30:00.000Z  News_Social      MP:4.0.0       956   
55    2014-11-11T20:30:00.000Z  News_Social      MP:4.0.0       948   
56    2014-11-12T20:30:00.000Z  News_Social      MP:4.0.0      1178   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

            buzz  sentiment  negative  positive  optimism  pessimism  \
52      2337.750   0.085980  0.194418  0.280398  0.079136   0.028660   
53      1465.500  -0.025589  0.258615  0.233026  0.036165   0.033095   
54      2921.450   0.023105  0.225231  0.248336  0.072567   0.056992   
55      2260.800   0.045780  0.196391  0.242171  0.067012   0.054406   
56      3298.050   0.108246  0.202999  0.311245  0.064280   0.032595   
...          ...        ...       ...       ...       ...        ...   
2984  116049.500  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.352  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.297  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.450  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.160  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
52    0.004919  0.004705 -0.014758  0.010266      0.027591  0.003422   
53    0.005118  0.007847 -0.018083  0.016718      0.029342  0.003753   
54    0.004621  0.003765 -0.009927  0.011980      0.030635  0.003252   
55    0.010616  0.008625 -0.000663  0.001106      0.014818  0.009731   
56    0.006216  0.016222  0.003790  0.004245      0.006822  0.005761   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
52    0.008983  0.033365  0.011763  0.001497     0.038926     0.015613   
53    0.024906  0.044353  0.006482  0.005459     0.016035     0.018424   
54    0.015746  0.038679  0.010269  0.004792     0.007702     0.028308   
55    0.010395  0.044453  0.021674  0.007962     0.016587     0.043259   
56    0.017738  0.036234  0.007883  0.006671     0.014706     0.019557   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0.003109    -

In [99]:
df_new = df_merge_Indicators_withoutDuplciates_1.dropna()

In [100]:
df_new

id assetcode  \
52    mp:2014-11-08_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
53    mp:2014-11-09_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
54    mp:2014-11-10_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
55    mp:2014-11-11_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
56    mp:2014-11-12_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...                                                 ...       ...   
2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
52    2014-11-08T20:30:00.000Z  News_Social      MP:4.0.0       925   
53    2014-11-09T20:30:00.000Z  News_Social      MP:4.0.0       759   
54    2014-11-10T20:30:00.000Z  News_Social      MP:4.0.0       956   
55    2014-11-11T20:30:00.000Z  News_Social      MP:4.0.0       948   
56    2014-11-12T20:30:00.000Z  News_Social      MP:4.0.0      1178   
...                        ...          ...           ...       ...   
2984  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2985  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2986  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2987  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2988  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

            buzz  sentiment  negative  positive  optimism  pessimism  \
52      2337.750   0.085980  0.194418  0.280398  0.079136   0.028660   
53      1465.500  -0.025589  0.258615  0.233026  0.036165   0.033095   
54      2921.450   0.023105  0.225231  0.248336  0.072567   0.056992   
55      2260.800   0.045780  0.196391  0.242171  0.067012   0.054406   
56      3298.050   0.108246  0.202999  0.311245  0.064280   0.032595   
...          ...        ...       ...       ...       ...        ...   
2984  116049.500  -0.090224  0.267235  0.177011  0.044300   0.056553   
2985   84260.352  -0.091645  0.263801  0.172157  0.040357   0.054753   
2986   67521.297  -0.090431  0.266494  0.176063  0.048007   0.062003   
2987  130180.450  -0.127907  0.285504  0.157597  0.040663   0.065233   
2988  147773.160  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
52    0.004919  0.004705 -0.014758  0.010266      0.027591  0.003422   
53    0.005118  0.007847 -0.018083  0.016718      0.029342  0.003753   
54    0.004621  0.003765 -0.009927  0.011980      0.030635  0.003252   
55    0.010616  0.008625 -0.000663  0.001106      0.014818  0.009731   
56    0.006216  0.016222  0.003790  0.004245      0.006822  0.005761   
...        ...       ...       ...       ...           ...       ...   
2984  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2985  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2986  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2987  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2988  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
52    0.008983  0.033365  0.011763  0.001497     0.038926     0.015613   
53    0.024906  0.044353  0.006482  0.005459     0.016035     0.018424   
54    0.015746  0.038679  0.010269  0.004792     0.007702     0.028308   
55    0.010395  0.044453  0.021674  0.007962     0.016587     0.043259   
56    0.017738  0.036234  0.007883  0.006671     0.014706     0.019557   
...        ...       ...       ...       ...          ...          ...   
2984  0.034925  0.048044  0.010681  0.004028    -0.001512     0.019463   
2985  0.041888  0.045579  0.007892  0.003109    -

In [101]:
2989-52

2937

In [102]:
df_new['Close']

52        345.488007
53        363.264008
54        366.924011
55        367.695007
56        423.561005
            ...     
2984    16697.777344
2985    16711.546875
2986    16291.832031
2987    15787.284180
2988    16189.769531
Name: Close, Length: 2937, dtype: float64

In [103]:
df_new['close']

52        345.488007
53        363.264008
54        366.924011
55        367.695007
56        423.561005
            ...     
2984    16697.777340
2985    16711.546880
2986    16291.832030
2987    15787.284180
2988    16189.769530
Name: close, Length: 2937, dtype: float64

In [104]:
df_new.iloc[2935]['Close']==df_new.iloc[2935]['close']

False

In [105]:
df_new.iloc[2935]['Close']

15787.2841796875

In [106]:
df_new.iloc[2935]['close']

15787.28418

In [107]:
df_new[df_new['close'].apply(int) != df_new['Close'].apply(int)]

Empty DataFrame
Columns: [id, assetcode, windowtimestamp, datatype, systemversion, mentions, buzz, sentiment, negative, positive, optimism, pessimism, joy, lovehate, trust, anger, disagreement, fear, gloom, stress, violence, surprise, timeurgency, uncertainty, emotionvsfact, shortvslongterm, marketrisk, longshort, longshortforecast, pricedirection, pricedown, priceforecast, priceup, pricemomentum, topvsbottom, overvaluedvsundervalued, volatility, adoption, adoptionforecast, anonymity, attack, codesentiment, codeupgrade, criminalactivity, developersentiment, fomo, fork, forkforecast, hodl, innovation, litigation, majorityattack, noobs, regulatorycrackdown, regulatoryissues, scam, transactionspeed, vulnerability, partnership, futurevspast, windowtimestamp1, windowtimestamp2, date, close, r_BTC_USD, R_BTC_USD, volume_BTC_USD, dateFormatted, Date, Open, High, Low, Close, Adj Close, Volume, MoneyFlowVolume, accumulation_distribution_indicator, Aroon_up, Aroon_down, Aroon_oscillator, Adj Open, Adj High, Adj Low, adx_pos_directional_indicator, adx_neg_directional_indicator, avg. Directional index, avg.True Range, upper_Bollinger_Bands, lower_Bollinger_Bands, simple_moving_avg_20, Chaikin money flow, chaikin_oscillator, Commodity Channel Index, Coppock Curve, Ease of Movement, Ease of Movement MA, ForceIndex, mass_index, MovingAverageConvergenceDivergence, MovingAverageConvergenceDivergence_signal, ...]
Index: []

Close feature in the data downloaded from yahooFinance and close from BTCPrice.csv are same

In [108]:
df_new.drop("Close",axis=1,inplace=True)

C:\Users\Student\AppData\Local\Temp\ipykernel_7772\2249017412.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new.drop("Close",axis=1,inplace=True)


In [109]:
df_new.reset_index(inplace=True)

In [110]:
df_new

index                                                 id assetcode  \
0        52  mp:2014-11-08_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1        53  mp:2014-11-09_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2        54  mp:2014-11-10_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3        55  mp:2014-11-11_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4        56  mp:2014-11-12_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...     ...                                                ...       ...   
2932   2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2933   2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2934   2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2935   2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2936   2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-11-08T20:30:00.000Z  News_Social      MP:4.0.0       925   
1     2014-11-09T20:30:00.000Z  News_Social      MP:4.0.0       759   
2     2014-11-10T20:30:00.000Z  News_Social      MP:4.0.0       956   
3     2014-11-11T20:30:00.000Z  News_Social      MP:4.0.0       948   
4     2014-11-12T20:30:00.000Z  News_Social      MP:4.0.0      1178   
...                        ...          ...           ...       ...   
2932  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2933  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2934  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2935  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2936  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

            buzz  sentiment  negative  positive  optimism  pessimism  \
0       2337.750   0.085980  0.194418  0.280398  0.079136   0.028660   
1       1465.500  -0.025589  0.258615  0.233026  0.036165   0.033095   
2       2921.450   0.023105  0.225231  0.248336  0.072567   0.056992   
3       2260.800   0.045780  0.196391  0.242171  0.067012   0.054406   
4       3298.050   0.108246  0.202999  0.311245  0.064280   0.032595   
...          ...        ...       ...       ...       ...        ...   
2932  116049.500  -0.090224  0.267235  0.177011  0.044300   0.056553   
2933   84260.352  -0.091645  0.263801  0.172157  0.040357   0.054753   
2934   67521.297  -0.090431  0.266494  0.176063  0.048007   0.062003   
2935  130180.450  -0.127907  0.285504  0.157597  0.040663   0.065233   
2936  147773.160  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.004919  0.004705 -0.014758  0.010266      0.027591  0.003422   
1     0.005118  0.007847 -0.018083  0.016718      0.029342  0.003753   
2     0.004621  0.003765 -0.009927  0.011980      0.030635  0.003252   
3     0.010616  0.008625 -0.000663  0.001106      0.014818  0.009731   
4     0.006216  0.016222  0.003790  0.004245      0.006822  0.005761   
...        ...       ...       ...       ...           ...       ...   
2932  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2933  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2934  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2935  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2936  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.008983  0.033365  0.011763  0.001497     0.038926     0.015613   
1     0.024906  0.044353  0.006482  0.005459     0.016035     0.018424   
2     0.015746  0.038679  0.010269  0.004792     0.007702     0.028308   
3     0.010395  0.044453  0.021674  0.007962     0.016587     0.043259   
4     0.017738  0.036234  0.007883  0.006671     0.014706     0.019557   
...        ...       ...       ...       ...          ...         

In [111]:
df_new.columns

Index(['index', 'id', 'assetcode', 'windowtimestamp', 'datatype',
       'systemversion', 'mentions', 'buzz', 'sentiment', 'negative',
       ...
       'Relative Strength Index', '%K-FAST Stochastic indicator',
       '%D- slow stochastic indicator', 'Trix', 'ulcer_index',
       'UltimateOscillator', 'VortexIndicator_diff', 'VortexIndicator_neg',
       'VortexIndicator_pos', 'WilliamsR'],
      dtype='object', length=121)

In [112]:
df_new['Aroon_up'].replace(0,0.0001,inplace=True)
df_new['Aroon_down'].replace(0,0.0001,inplace=True)
df_new['Aroon_oscillator'].replace(0,0.0001,inplace=True)
df_new['%K-FAST Stochastic indicator'].replace(0,0.0001,inplace=True)
df_new['ulcer_index'].replace(0,0.0001,inplace=True)
df_new['WilliamsR'].replace(0,0.0001,inplace=True)


C:\Users\Student\AppData\Local\Temp\ipykernel_7772\1576713547.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new['Aroon_up'].replace(0,0.0001,inplace=True)
C:\Users\Student\AppData\Local\Temp\ipykernel_7772\1576713547.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_new['Aroon_down'].replace(0,0.0001,inplace=True)
C:\Users\Student\AppData\Local\Temp\ipykernel_7772\1576713547.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-co

In [113]:
df_new.to_csv("BTCPriceCleaned.csv")

In [114]:
df_new

index                                                 id assetcode  \
0        52  mp:2014-11-08_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
1        53  mp:2014-11-09_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2        54  mp:2014-11-10_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
3        55  mp:2014-11-11_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
4        56  mp:2014-11-12_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
...     ...                                                ...       ...   
2932   2984  mp:2022-11-18_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2933   2985  mp:2022-11-19_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2934   2986  mp:2022-11-20_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2935   2987  mp:2022-11-21_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   
2936   2988  mp:2022-11-22_20.30.00.News_Social.CRYPTO.BITCOIN   BITCOIN   

               windowtimestamp     datatype systemversion  mentions  \
0     2014-11-08T20:30:00.000Z  News_Social      MP:4.0.0       925   
1     2014-11-09T20:30:00.000Z  News_Social      MP:4.0.0       759   
2     2014-11-10T20:30:00.000Z  News_Social      MP:4.0.0       956   
3     2014-11-11T20:30:00.000Z  News_Social      MP:4.0.0       948   
4     2014-11-12T20:30:00.000Z  News_Social      MP:4.0.0      1178   
...                        ...          ...           ...       ...   
2932  2022-11-18T20:30:00.000Z  News_Social     MP:4.0.22     30130   
2933  2022-11-19T20:30:00.000Z  News_Social     MP:4.0.22     22790   
2934  2022-11-20T20:30:00.000Z  News_Social     MP:4.0.22     19882   
2935  2022-11-21T20:30:00.000Z  News_Social     MP:4.0.22     34228   
2936  2022-11-22T20:30:00.000Z  News_Social     MP:4.0.22     34889   

            buzz  sentiment  negative  positive  optimism  pessimism  \
0       2337.750   0.085980  0.194418  0.280398  0.079136   0.028660   
1       1465.500  -0.025589  0.258615  0.233026  0.036165   0.033095   
2       2921.450   0.023105  0.225231  0.248336  0.072567   0.056992   
3       2260.800   0.045780  0.196391  0.242171  0.067012   0.054406   
4       3298.050   0.108246  0.202999  0.311245  0.064280   0.032595   
...          ...        ...       ...       ...       ...        ...   
2932  116049.500  -0.090224  0.267235  0.177011  0.044300   0.056553   
2933   84260.352  -0.091645  0.263801  0.172157  0.040357   0.054753   
2934   67521.297  -0.090431  0.266494  0.176063  0.048007   0.062003   
2935  130180.450  -0.127907  0.285504  0.157597  0.040663   0.065233   
2936  147773.160  -0.151787  0.302152  0.150366  0.039716   0.064626   

           joy  lovehate     trust     anger  disagreement      fear  \
0     0.004919  0.004705 -0.014758  0.010266      0.027591  0.003422   
1     0.005118  0.007847 -0.018083  0.016718      0.029342  0.003753   
2     0.004621  0.003765 -0.009927  0.011980      0.030635  0.003252   
3     0.010616  0.008625 -0.000663  0.001106      0.014818  0.009731   
4     0.006216  0.016222  0.003790  0.004245      0.006822  0.005761   
...        ...       ...       ...       ...           ...       ...   
2932  0.005377  0.001991 -0.002206  0.011564      0.031870  0.005519   
2933  0.006361  0.001727 -0.004925  0.011654      0.035551  0.004848   
2934  0.007176  0.002577 -0.001777  0.012041      0.031657  0.005376   
2935  0.005823  0.001575 -0.002255  0.012114      0.024312  0.007052   
2936  0.005366  0.001387 -0.003319  0.011014      0.023032  0.007505   

         gloom    stress  violence  surprise  timeurgency  uncertainty  \
0     0.008983  0.033365  0.011763  0.001497     0.038926     0.015613   
1     0.024906  0.044353  0.006482  0.005459     0.016035     0.018424   
2     0.015746  0.038679  0.010269  0.004792     0.007702     0.028308   
3     0.010395  0.044453  0.021674  0.007962     0.016587     0.043259   
4     0.017738  0.036234  0.007883  0.006671     0.014706     0.019557   
...        ...       ...       ...       ...          ...         